proxy/src/set_public_ip_and_start.sh

```diff
@@ -31,9 +31,28 @@ then
     fi
 fi
 
+## PUBLIC_IP retrieved from third-party sources
+if [[ $PUBLIC_IP == '' ]]
+then
+    urls=(
+        'https://icanhazip.com/'
```

Comment:
Won't this return the port too? Can you please test your change and verify all these options work and update the PR?

Reply:
I have tested them actually, no issues with ports. These sites return the IP in plain text and as I used them previously I never encountered such thing.

But it's fine, I'll double check and update the PR if necessary.

Reply:
Let's do not hardcode this. Do not make a bottle neck. What if icanhazip down? DoS? Blocked/filtered?

Reply:
@slawlor I tested them both on my local machine by swapping the order of URLs and on remote servers with different OSs, they all worked fine with no port ever causing issues, they only return IPs. I think we are good to go.

Reply:
@FFAMax Currently the PUBLIC_IP is entirely manual if you are not on AWS, so this actually solves a problem instead of creating one, and I don’t think of a way except hardcoding it.

In case one of the services not being available, there are two others that can resolve it.

BTW, the first one belongs to **Cloudflare** and is usually used as a source to get the IP and it is not blocked in most internet-restricted countries.

Reply:
This is still a problem, your loop never exits and will therefore always take the last IP address, valid or not.

Reply:
@slawlor yes finally I got it fixed.
The first issue as you mentioned was that it was always taking the last IP, the `&& break` at the end got it solved.
Then I also noticed that in the case of HTTP 4xx, and 5xx codes it was returning them. So I configured the curl to fail when in case of non-200 responses.

At last, as the flags were long and could be used in the original AWS IP resolve part, I created a function to wrap it all.

I tested it with multiple cases, such as the URL is blocked by the network, and the URL returns a 500 error, with various ordering of the URLs, seems I covered them all.

Let me know if there are any more improvements or issues left.

Reply:
Apologies in the delay. I haven't had time to do reviews, 1 small change requested and I think it's good to go!

Reply:
Just updated it. it should be good now.